spec/ddtrace/contrib/rails/rack_spec.rb

```diff
@@ -7,6 +7,7 @@
   let(:routes) do
     {
       '/full' => 'test#full',
+      '/partial' => 'test#partial',
```

Comment:
Partial template rendering was the main feature that was tested in `controller_test.rb` that was still not covered by `rack_spec.rb`.

Other changes are additional assertions and a few fixes to already existing tests.